### PR TITLE
#19 Add equality

### LINKED ISSUE #19: Implement equality for CvParam to compare with another CvParam (StructuralEquality)

### PR DESCRIPTION
**Pull request by omaus**
This PR
- overrides standard `.Equals()` method for
  - UserParam
  - CvParam
  - CvContainer
- adds respective unit tests to CvParam
- overrides standard `.GetHashCode()` method as adviced by the compiler for
  - UserParam
  - CvParam
  - CvObject
  - CvContainer
- closes #19

### CODE REVIEW

File: tests/ControlledVocabulary.Tests/CvParamTests.fs [hunk 1]
```diff
@@ -30,6 +34,13 @@ module InstanceMemberTests =
         let actual = testCvParams |> List.map (fun x -> x.Value)
         Assert.Equal<ParamValue List>(expected, actual)
 
+    [<Fact>]
+    let ``Equals`` () =
+        let testCvp1 = CvParam("test", "test", "test", ParamValue.Value "test", Generic.Dictionary<string,IParam>() |> fun d -> d.Add("test", testCvParams.Head); d) 
+        let testCvp2 = CvParam("test", "test", "test", ParamValue.Value "test", Generic.Dictionary<string,IParam>() |> fun d -> d.Add("test", testCvParams.Head); d)
```
Comment (kMutagene) [inline tests/ControlledVocabulary.Tests/CvParamTests.fs:40]: Can you add some tests for CvParams with attributes? I am thinking of one where we expect them to be equal, and one where we expect them not to be equal

**Review by kMutagene — CHANGES_REQUESTED** [review]
Thanks! I think we'll need some more tests to be sure, especially regarding the attribute collection. We should keep in mind that this replaces the test utility functions for cvparam equality used in ARCTokenization, bu i'll open a separate issue for that.

File: tests/ControlledVocabulary.Tests/CvParamTests.fs (hunk 1 above)
Comment (kMutagene) [reply tests/ControlledVocabulary.Tests/CvParamTests.fs:40]: Sorry, clarification: Attributes where the attributes themselves have attributes

**Comment by omaus** [comment]
Added more unit tests as requested. 👍🏻

**Review by kMutagene — APPROVED** [review]

**Pull request merged**